addons/controllers/awsebscsi/awsescsiconfig_utils.go

```diff
@@ -0,0 +1,61 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	clusterapiv1beta1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/vmware-tanzu/tanzu-framework/addons/pkg/constants"
+	csiv1alpha1 "github.com/vmware-tanzu/tanzu-framework/apis/addonconfigs/csi/v1alpha1"
+)
+
+// getOwnerCluster verifies that the AwsEbsCSIConfig has a cluster as its owner reference,
+// and returns the cluster. It tries to read the cluster name from the AwsEbsCSIConfig's owner reference objects.
+// If not there, we assume the owner cluster and AwsEbsCSIConfig always has the same name.
+func (r *AwsEbsCSIConfigReconciler) getOwnerCluster(ctx context.Context,
+	awsebsCSIConfig *csiv1alpha1.AwsEbsCSIConfig) (*clusterapiv1beta1.Cluster, error) {
+
+	logger := log.FromContext(ctx)
+	cluster := &clusterapiv1beta1.Cluster{}
+	// TODO donot support using addon name as cluster name
+	clusterName := awsebsCSIConfig.Name
+
+	// retrieve the owner cluster for the AwsEbsCSIConfig object
+	for _, ownerRef := range awsebsCSIConfig.GetOwnerReferences() {
```

Comment:
If an user creates an following AwsEbsCSIConfig, can this method return the corresponding Cluster. 

- with a name not match the name convention
- without a Cluster owner reference 
- referred in the corresponding ClusterBootsrap

Reply:
- with a name not match the name convention -> Vija will have a coming change to support random name, ideally, we will use owner reference go get cluster name, hopefully current code 
```
for _, ownerRef := range awsebsCSIConfig.GetOwnerReferences() {
		if strings.EqualFold(ownerRef.Kind, constants.ClusterKind) {
			clusterName = ownerRef.Name
			break
		}
	}
```
could handle this, I still left a TODO in case, we will need a coming PR to adpot Vija's changes.